src/catch2/matchers/catch_matchers_elements_are.hpp

```diff
@@ -8,61 +8,124 @@
 #ifndef CATCH_MATCHERS_ELEMENTS_ARE_HPP_INCLUDED
 #define CATCH_MATCHERS_ELEMENTS_ARE_HPP_INCLUDED
 
-#include <catch2/matchers/catch_matchers_templated.hpp>
-
 #include <algorithm>
+#include <catch2/matchers/catch_matchers_templated.hpp>
 #include <utility>
 
 namespace Catch {
-namespace Matchers {
-
-    template<typename RangeLike>
-    class ElementsAreElementsMatcher final : public MatcherGenericBase {
-        RangeLike m_range;
-	public:
-		ElementsAreElementsMatcher(RangeLike range):
-		    m_range(std::move(range))
-		{}
-
-		bool match(RangeLike rng) const {
-		    return std::equal(m_range.begin(), m_range.end(), rng.begin());
-		}
-
-		std::string describe() const override {
-		    return "elements are " + Catch::Detail::stringify(m_range);
-		}
-    };
-
-    template<typename RangeLike>
-    class UnorderedElementsAreElementsMatcher final : public MatcherGenericBase {
-        RangeLike m_range;
-	public:
-		UnorderedElementsAreElementsMatcher(RangeLike range):
-		    m_range(std::move(range))
-		{}
-
-		bool match(RangeLike rng) const {
-		    return std::is_permutation(m_range.begin(), m_range.end(), rng.begin());
-		}
-
-		std::string describe() const override {
-		    return "unordered elements are " + ::Catch::Detail::stringify( m_range );
-		}
-    };
-
-    //! Creates a matcher that checks if all elements in a range are equal to all elements in another range
-    template<typename RangeLike>
-    ElementsAreElementsMatcher<RangeLike> ElementsAre(RangeLike rng) {
-        return { std::forward<RangeLike>(rng) };
-    }
-
-    //! Creates a matcher that checks if all elements in a range are equal to all elements in another range in some permutation
-    template<typename RangeLike>
-    UnorderedElementsAreElementsMatcher<RangeLike> UnorderedElementsAre(RangeLike rng) {
-        return { std::forward<RangeLike>(rng) };
-    }
-
-} // namespace Matchers
+    namespace Matchers {
```

Comment:
huh, is this actually the formatting given by clang-format?

I would swear the clang-format had nested namespace indents disabled.

Reply:
It seems not: `NamespaceIndentation: All` is specified (should be `None` for zero indent) :https://github.com/catchorg/Catch2/blob/61d0f7a9afb88c12f8931f80321e58f2cfbc62db/.clang-format#L16